doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,892 @@
+                description "Current load max value.";
+            }
+
+            leaf ipv4_enable {
+                type boolean;
+                description "Whether ARS is enabled over IPv4 packets";
+            }
+
+            leaf ipv6_enable {
+                type boolean;
+                description "Whether ARS is enabled over IPv6 packets";
+            }
+        }
+        /* end of list ARS_PROFILE_LIST */
+    }
+    /* end of container ARS_PROFILE */
+```
+
+##### ARS_INTERFACE
+
+```
+    container ARS_INTERFACE {
+
+        list ARS_INTERFACE_LIST {
+            description  "List of interfaces participating in ARS";
+            key "if_name";
+
+            leaf if_name {
+                type leafref {
+                    path "/port:sonic-port/port:PORT/port:PORT_LIST/port:name";
+                }
+                description "ARS-enabled interface name";
+            }
+
+            leaf scaling_factor {
+                type uint32;
+                default "10000";
+                description "This factor used to normalize load measurements across ports with different speeds.";
+            }
+
+        }
+        /* end of list ARS_INTERFACE_LIST */
+    }
+    /* end of container ARS_INTERFACE */
+```
+
+##### ARS_NEXTHOP_GROUP
+
+```
+    container ARS_NEXTHOP_GROUP {
+
+        description "ARS-enabled Nexthop Groups";
+
+        list ARS_NEXTHOP_GROUP_LIST {
+
+            key "vrf_name ip_prefix";
+
+            leaf vrf_name {
+                type union {
+                    type string {
+                        pattern "default";
+                    }
+                    type leafref {
+                        path "/vrf:sonic-vrf/vrf:VRF/vrf:VRF_LIST/vrf:name";
+                    }
+                }
+                description "VRF name";
+            } 
+
+            leaf ip_prefix{
+                type stypes:sonic-ip-prefix;
+                description "Ip prefix which identifies nexthop group for which ARS behavior is desired";
+            }
+
+            leaf profile_name {
+                description "ARS profile Name";
+                type leafref {
+                    path "/sars:sonic-ars/sars:ARS_PROFILE/sars:ARS_PROFILE_LIST/sars:profile_name";
+                }
+            }
+
+            leaf assign_mode {
+                type enumeration {
+                    enum per_flowlet_quality{
+                        description "Per flow-let assignment based on flow quality";
+                    }
+                    enum per_packet {
+                        description "Per packet flow assignment based on port load";
+                    }
+                }
+            }
+
+            leaf flowlet_idle_time {
+                type uint16 {
+                    range 2..2047;
+                }
+                description  "Idle duration in microseconds. This duration is to classifying a flow-let in a macro flow.";
+            }
+
+            leaf max_flows {

```

Comment:
max_flows is a field in the ars_profile table as well. Is this a duplicate or the intent here is to create a separate limit per ip-prefix?